### PR TITLE
Med 90 implement patient card api

### PR DESCRIPTION
**Pull request by skazzp**
**Patients card API:**

- Added route to get/update patient info
- Added routes to add/remove appointment notes
- Added get route for notes with search, pagination and sort

1.2 Backend

- [x] 1.2.1 swagger for each endpoint. add in the documentation, different types of responses.
 on a successful response (2xx), on unsuccessful response (4xx, 5xx)

- [x] 1.2.2 less requests to db, all data should be taken with one query
- [x] 1.2.3 `public` in method use only is use function externally
- [x] 1.2.4 use ConfigService instead of `process.env`
- [x] 1.2.5 use @`@index` decorator  for frequently requested data
- [x] 1.2.6 use transactions if there is a call chain that mutates data in different tables
- [x] 1.2.7. add a set of rules for [[nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed)](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments

### CODE REVIEW

**Review by Mati-Pereira — APPROVED** [review]

File: src/modules/patient-notes/patient-notes.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    private notesRepo: Repository<PatientNotes>,
+  ) {}
+
+  async addPatientNote(dto: CreatePatientNoteDto): Promise<PatientNotes> {
+    const note = await this.notesRepo.save(dto);
+
+    return note;
+  }
+
+  async getNotes(id: number, query: NotesSearchOptionsDto): Promise<NotesRes> {
```
Comment (lyudad) [inline src/modules/patient-notes/patient-notes.service.ts]: pass to function only data that needed, query is bad example of naming

File: src/modules/patients/patients.service.ts [hunk 2]
```diff
@@ -23,27 +24,59 @@ export class PatientsService {
...
+  }
+
+  async updatePatient(id: number, payload: UpdatePatientDto): Promise<Patient> {
+    try {
+      const patient = await this.repo
+        .createQueryBuilder('patient')
+        .where('id = :id', { id })
+        .getOne();
+
+      if (!patient) throw new UnauthorizedException('User not found!');
```
Comment (lyudad) [inline src/modules/patients/patients.service.ts]: find the way to make check and update in the same time

File: src/modules/patient-notes/patient-notes.service.ts (hunk 1 above)
Comment (skazzp) [reply src/modules/patient-notes/patient-notes.service.ts]: renamed

File: src/modules/patients/patients.service.ts (hunk 2 above)
Comment (skazzp) [reply src/modules/patients/patients.service.ts]: Made changes, removed "find" request to db

File: src/modules/patients/patients.controller.ts [hunk 3]
```diff
@@ -40,20 +43,62 @@ export class PatientsController {
...
+  })
+  @ApiResponse({
+    status: 401,
+    description: 'Patient not found',
+  })
+  async confirm(
+    @Param() params: { id: number },
+    @Body() dto: UpdatePatientDto,
+  ): Promise<IServerResponse<UpdatePatientDto>> {
+    const patient = await this.patientsService.updatePatient(+params.id, dto);
```
Comment (lyudad) [inline src/modules/patients/patients.controller.ts]: instead of doing +params.id all the time you can create a general rule for it

Comment (skazzp) [reply src/modules/patients/patients.controller.ts]: It was working without "+" too, forgot to remove it.
Not sure what do you meant by general rule, but added class-transformer @Type(() => Number) to validation

File: src/modules/patient-notes/patient-notes.service.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    const response = await queryBuilder
+      .take(limit)
+      .skip(skip)
+      .where('notes.patient_id = :id', { id })
+      .leftJoin('notes.doctor', 'doctor')
+      .addSelect(['doctor.firstName', 'doctor.lastName'])
+      .orderBy('notes.updatedAt', order)
+      .getManyAndCount();
```
Comment (lyudad) [inline src/modules/patient-notes/patient-notes.service.ts:51]: I think you can set condition to one line where is difference instead of full query

Comment (skazzp) [reply src/modules/patient-notes/patient-notes.service.ts:51]: made changes

**Review by lyudad — APPROVED** [review]

**Pull request merged**